samples: display: lvgl: test with XIAO round display

mniestroj opened the pull request:
"XIAO round display" shield is compatible with most XIAO boards and is very nice platform for using
as round smartwatch-like display. Add tests of LVGL sample based on selected XIAO boards in order to
make sure this shield definition remains valid.

mniestroj committed:
samples: display: lvgl: increase main stack size to 4k

When tested with xiao_esp32s3 + seeed_xiao_round_display main stack is
occupied with 3504 bytes (checked with 'kernel thread stacks' command).

Increase main stack size from 2k to 4k in order to make this sample work at
runtime.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

mniestroj committed:
samples: display: lvgl: test sample with 'seeed_xiao_round_display'

This shield is compatible with all XIAO boards. Tested with 2 of them:
 * xiao_ble
 * xiao_esp32s3

Extend sample.yaml for integration testing with both, so there is coverage
across two very distinct SoC vendors and architectures.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

mniestroj committed:
boards: seeed: add xiao_{adc,gpio,i2c,serial,spi} as supported

Mark interfaces on XIAO header that are supported by each board. This
allows to filter by them (with depends_on) when running twister.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

kartben commented on samples/subsys/display/lvgl/sample.yaml:
```diff
@@ -108,3 +108,26 @@ tests:
+  samples.subsys.display.lvgl.seeed_xiao_round_display:
+    min_flash: 320
+    min_ram: 64
+    depends_on:
```
@JarmouniA what do you mean by non-standard?

JarmouniA replied:
Sorry `non-standard` is not really a useful description, I meant to say very platform-specific. I don't think it would be good to start adding twister tags for every kind of boards, it would be very confusing... but that's just my opinion.

mniestroj replied:
Introduced `xiao_{adc,i2c,spi,serial,gpio}` are supposed to map what we have for `arduino_{i2c,spi,serial,gpio}` already. The nice thing about it is that all boards that have the same header are ready to be tested without specifying each board manually inside `platform_allow`.

JarmouniA replied:
Arduino headers are a universal thing that can be found on a lot of dev board, that's why it has its own tags.

mniestroj commented on samples/subsys/display/lvgl/sample.yaml:
```diff
@@ -108,3 +108,26 @@ tests:
+      - xiao_i2c
+      - xiao_spi
+    tags:
+      - samples
```
Each test has a bit different set of tags, we don't use the same tags there for each test.

JarmouniA replied:
`samples display gui lvgl` are common to all test cases, `shield` can be left in test cases that need it.

mniestroj committed:
samples: display: lvgl: deduplicate sample.yaml using 'common' section

Move 'tags' and 'modules' into common section in samples.yaml, so test
definitions are deduplicated a bit.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

mniestroj committed:
samples: display: lvgl: s/samples/sample/ in sample.yaml

Use 'sample.' in the sample definition, in order to follow Zephyr
convention.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

mniestroj commented on samples/subsys/display/lvgl/sample.yaml:
```diff
@@ -1,6 +1,15 @@
+    - gui
+    - lvgl
+    - samples
+    - shield
```
Yes, noticed the same and just fixed that.
